pinot-core/src/main/java/org/apache/pinot/core/transport/ServerThreadPoolConfig.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.transport;
+
+public class ServerThreadPoolConfig {
```

Comment:
1) This is just for the threads for the http server used in server/broker/controller. Can we make the class name more explicit on this?
2) Add javadoc

Reply:
Renamed to `HttpServerThreadPoolConfig` and added a javadoc. Lmk your thoughts.

Reply:
Can you rename controller.thread.pool.corePoolSize to controller.httpserver.thread.pool.corePoolSize because it is about http servers running on controller/broker/server?

Reply:
Good point. Updated to `controller.http.server.thread.pool.corePoolSize`.